### DISCUSSION TITLE
Supabase-js - "Property 'catch' does not exist on type 'PromiseLike<void>'"

### DISCUSSION THREAD

**Discussion by taValendo** [supabase/supabase#5047] [Questions]
Hey!

I always get this TS error:
`Property 'catch' does not exist on type 'PromiseLike<void>`
when trying to use `.catch` after invoking `supabase.from(...).select(...).then()`. Am I missing something? I think `PromiseLike` should include `.catch` and `.finally`.

**Comment by GaryAustin1** [comment]
These two threads might give you an idea, but I believe Supabase attempts to pass the errors back directly only.

https://github.com/supabase/supabase-js/issues/92
https://github.com/supabase/postgrest-js/pull/188

Appears they added a .throwOnError() method to get closer to the .catch type of operation.  But I've never found clear documentation on it myself.

**Reply by taValendo** [reply]
thanks @GaryAustin1

**Comment by Xananax** [comment]
For one project, using `.catch()` is neater, so I wrap all my API like this:

```ts
import { PostgrestSingleResponse } from "@supabase/supabase-js";
import { supabase } from "./supabase";

const handleSupabaseResponse = <T>({error, data}: PostgrestSingleResponse<T>) => {
  if (error) { throw Error;}
  return data;
};

const wrapper = <T, P extends PostgrestSingleResponse<T>>(s: PromiseLike<P>) => Promise.resolve(s);

export const getPostById = (post_id: number) =>
  wrapper(supabase.rpc("get_thread_tree_by_id", { post_id })).then(
    handleSupabaseResponse
  );

export const getPostsListBySlug = (post_slug: string) =>
  wrapper(supabase.rpc("get_thread_trees_by_slug", { post_slug })).then(
    handleSupabaseResponse
  );
```

Then I can just do

```ts
getPostById(1).then(...).catch(...)
```
And it remains well typed.

There's probably a way to make `wrapper` contain the `handleSupabaseResponse` in one go instead of repeating `.then(...)` in each command, but I wasn't able to yield Typescript to do that, and this works.
